composer/datasets/streaming/dataset.py

```diff
@@ -90,6 +92,10 @@ class StreamingDataset(IterableDataset):
             to convert the raw bytes to an object.
         max_retries (int): Number of download re-attempts before giving up. Default: 2.
         timeout (float): How long to wait for shard to download before raising an exception. Default: 60 sec.
+        min_buffer (int): During the first epoch, the minimum number of samples that must be available before it
```

Comment:
Thanks for doing this too, but could you put it in a follow-on PR? There are a few ways to "limit" the amount of data that is greedily downloaded -- one is max samples, another is prefetch factor, another is max size. I'm personally am a fan of  a prefetch factor argument, to match pytorch, and because it could remain the same regardless of sample size or number of samples, but would like to take a quick straw poll first.